### ISSUE TITLE
Question about stream created on top of stream created through schema registry stream (JSON format) 

### ISSUE THREAD

**Issue by eshil-patel** [confluentinc/ksql#6704]
Setup: 
Ksqldb version 0.11.0
schema registry version 5.5.1 (from cp-all-in-one)

I have a question/scenario that is similar to this issue : https://github.com/confluentinc/ksql/issues/6435 . 
I run this query first (with a schema in schema registry named xyz-value), and it creates a stream successfully 
```
CREATE stream baseStream WITH (KAFKA_TOPIC = 'xyz',VALUE_FORMAT='JSON');
```
Then I run this query , whihc successfully creates a stream 
```
CREATE stream testStream as select field -> subfield -> subfield -> subfield -> fieldValue from BASESTREAM
```
However, I am unable to get any data through testStream, even though I see data flowing through baseStream. I am having trouble understanding what solution was posed in the issue I linked, or if there is another way to get data flowing through testStream. Any help would be greatly appreciated !

**Comment by vpapavas** [comment]
Hi @eshil-patel ,

I think the problem is due to the way the messages are produced. It seems to me that they are produced using JSON format  (is that what org.springframework.kafka.support.serializer.JsonSerializer produces?) which is not byte-compatible with JSON_SR format. Since you are creating a stream with JSON_SR format, you are telling ksql to expect a magic byte (0x00) and then four bytes indicating the schema id (0x00 0x00 0x00 0x01 for example) for the data it is reading. However, the data have not been encoded that way since they are produced in JSON.

**Comment by eshil-patel** [comment]
Thank you for the response. It looks like the class I was originally using did serialize it into JSON, and not JSON_SR. I was wondering if the following is possible : Have a consumer read a message in JSON, and then have a producer write that message in JSON_SR ? Thanks for the help.

**Comment by vpapavas** [comment]
Yes, I think that should work. You can take a look here for how to produce JSON_SR to a topic https://www.confluent.io/blog/confluent-platform-now-supports-protobuf-json-schema-custom-formats/
